docs/tutorials/HowToCreateAProduct.md

```diff
@@ -0,0 +1,92 @@
+All ProductTypes are derived from `ProductType`. The methode `Instantiate()` returns an object of the correspoding `ProductInstance`.
+
+With the attibute `DisplayName` you can change the name displayed on the UI. Using `Description` adds descriptions to a property on the UI. All public properties of a `ProductType` will be displayed on the UI.
+```C#
+[DisplayName("Watch Needle")]
+public class NeedleType : ProductType
+{
+    // Properties of the ProductType
+    [Description("Length of the needle")]
+    public int Length { get; set; }
+
+    // Creates the corresponding instance of the productType
+    protected override ProductInstance Instantiate()
+    {
+        return new NeedleInstance();
+    }
+}
+```
+All ProductInstances are derived from `ProductInstance<TProductType>`. Instance specific properties can be added. If you have for example only one materialnumber for a car type and the configuration depends on the user, this could be part of the properties of the instance.
+```C#
+public class NeedleInstance : ProductInstance<NeedleType>
+{
+    // Instance specific properties
+    public NeedleRole Role { get; set; }
+
+    ...
+}
+```
+
+## Use PartLinks
+Some `ProductTypes` have components. These can be modeled using [PartLinks](xref:Moryx.AbstractionLayer.Products.ProductPartLink). If no additional properties are needed, use the basic `ProductPartLink`. This is done for the part *WatchFace* in the example.
+
+```C#
+[DisplayName("Watch")]
+public class WatchType : ProductType
+{
+    // References to product
+    [DisplayName("Watch face")]
+    public ProductPartLink<WatchFaceType> WatchFace { get; set; }
+
+    // References to product using costum part link
+    [DisplayName("Watch needle")]
+    public List<NeedlePartLink> Needles { get; set; } = new List<NeedlePartLink>();
+
+    protected override ProductInstance Instantiate()
+    {
+        return new WatchInstance
+        {
+            //All parts need to be instantiated
+            WatchFace = (WatchFaceInstance)WatchFace.Instantiate(),
+            Needles = Needles.Instantiate<NeedleInstance>()
+        };
+    }
+}
+```
+If you want to add properties to the `PartLink`, you can create custom ones. In our example we used the costum part link `NeedlePartLink` to specify the role of the needle in the watch.
+```C#
+public class NeedlePartLink : ProductPartLink<NeedleType>
+{
+    public NeedlePartLink()
+    {
+    }
+
+    public NeedlePartLink(long id) : base(id)
+    {
+    }
+
+    // Partlink specific property
+    [Description("Which role does the needle have")]
+    public NeedleRole Role { get; set; }
+
+    public override ProductInstance Instantiate()
+    {
+        var needle = (NeedleInstance) base.Instantiate();
+        // Copy the partlink specific property to the instance
+        needle.Role = Role;
+        return needle;
+    }
+}
+```
+In `Instantiate()` we set the role of the instance to the role speficied in the PartLink by the user.
```

Comment:
Should there also be a tutorial for how to write an product importer or resource initializer? 🤔 that seems to be an essential part of resources and products in a project as well, or what would you say? 🤔